### DISCUSSION TITLE
Passing a HTML element/reference with Cypress

### DISCUSSION THREAD

**Discussion by muniz95** [cypress-io/cypress#29660] [Questions and Help]
Hello everyone.

I'm setting a component test class for an Angular component that accepts a string/TemplateRef as @Input:
`
@Input('cardHeader') cardHeader: string | TemplateRef<any> = '';
`

In my test case I can evaluate the test passing a string without any problem:
`
cy.mount(CardComponent, {
	componentProperties: {
		cardHeader: 'Test'
	}
})
`

I saw that TestBed has a method to create an instance for a given component; is there anything similar in Cypress? And if no, how can I pass an element to a component?

**Comment by portyu9** [comment]
Hi @muniz95 

The key point is that `TemplateRef` is **not an HTML element** that you can construct and pass directly through `componentProperties`.

Angular creates a `TemplateRef` from an `<ng-template>` during template compilation. For this case, mount a small host template that creates the template reference and passes it to your component:

```ts
cy.mount(`
  <ng-template #header>Test</ng-template>
  <your-card-selector [cardHeader]="header"></your-card-selector>
`, {
  declarations: [CardComponent],
})
```

Replace `your-card-selector` with the actual selector of `CardComponent`.

For a standalone Angular component, import the component rather than putting it in `declarations`.

So unlike the string case, the `TemplateRef` should be created by Angular's template system rather than manually instantiated in Cypress.

Please mark this answer as accepted if it helped, thank you.
